rosetta/services/mempool.go

```diff
@@ -0,0 +1,85 @@
+package services
+
+import (
+	"context"
+
+	"github.com/coinbase/rosetta-sdk-go/server"
+	"github.com/coinbase/rosetta-sdk-go/types"
+	ethCommon "github.com/ethereum/go-ethereum/common"
+	hmyTypes "github.com/harmony-one/harmony/core/types"
+	"github.com/harmony-one/harmony/hmy"
+	"github.com/harmony-one/harmony/rosetta/common"
+)
+
+// MempoolAPI implements the server.MempoolAPIServicer interface
+type MempoolAPI struct {
+	hmy *hmy.Harmony
+}
+
+// NewMempoolAPI creates a new instance of MempoolAPI
+func NewMempoolAPI(hmy *hmy.Harmony) server.MempoolAPIServicer {
+	return &MempoolAPI{
+		hmy: hmy,
+	}
+}
+
+// Mempool ...
+func (s *MempoolAPI) Mempool(
+	ctx context.Context, req *types.NetworkRequest,
+) (*types.MempoolResponse, *types.Error) {
+	if err := assertValidNetworkIdentifier(req.NetworkIdentifier, s.hmy.ShardID); err != nil {
+		return nil, err
+	}
+
+	pool, err := s.hmy.GetPoolTransactions()
+	if err != nil {
+		return nil, common.NewError(common.CatchAllError, map[string]interface{}{
+			"message": "unable to fetch pool transactions",
+		})
+	}
+	txIDs := make([]*types.TransactionIdentifier, pool.Len())
+	for i, tx := range pool {
+		txIDs[i] = &types.TransactionIdentifier{
+			Hash: tx.Hash().String(),
+		}
+	}
+	return &types.MempoolResponse{
+		TransactionIdentifiers: txIDs,
+	}, nil
+}
+
+// MempoolTransaction ...
+func (s *MempoolAPI) MempoolTransaction(
+	ctx context.Context, req *types.MempoolTransactionRequest,
+) (*types.MempoolTransactionResponse, *types.Error) {
+	if err := assertValidNetworkIdentifier(req.NetworkIdentifier, s.hmy.ShardID); err != nil {
+		return nil, err
+	}
+
+	hash := ethCommon.HexToHash(req.TransactionIdentifier.Hash)
+	poolTx := s.hmy.GetPoolTransaction(hash)
+	if poolTx == nil {
+		return nil, &common.TransactionNotFoundError
+	}
+
+	var nilAddress ethCommon.Address
+	estReceipt := &hmyTypes.Receipt{
+		PostState:         []byte{},
+		Status:            hmyTypes.ReceiptStatusSuccessful, // Assume transaction will succeed
+		CumulativeGasUsed: poolTx.Gas(),
+		Bloom:             [256]byte{},
+		Logs:              []*hmyTypes.Log{},
```

Comment:
Missing logs here for collect reward transaction will cause an error, reference:
https://github.com/harmony-one/harmony/blob/e74ab0bc0aa2d2fd39d7dadaa96ca5a5d4777fe9/rosetta/services/block.go#L1004-L1024

Can just create some dummy logs with the collect reward topic and have the value at 0.